src/vm/internal/compiler/lex.cc

```diff
@@ -841,10 +846,79 @@ static int skip_to(const char *token, const char *atoken) {
   }
 }
 
+void init_include_path() {
+  push_malloced_string(add_slash(current_file));        // does master has an include path?
+  svalue_t *ret = apply_master_ob(APPLY_GET_INCLUDE_PATH, 1);
+
+  if (!ret || ret == (svalue_t *)-1) {                  // either no or no master yet
+      return;                                           // just use the runtime configuration
+  }
+  if (ret->type != T_ARRAY) {                           // illegal return value
+      debug_message("'master::get_include_path' must return 'string *'\n");
+      return;                                           // we still have the runtime configuration
+  }
+  array_t *arr  = ret->u.arr;
+  int size = arr->size;
+
+  if(!size) {                                           // empty path?
+      debug_message("got empty include path for 'master::get_include_path(%s)'\n", current_file);
+      return;                                           // we still have the runtime configuration
+  }
+  char **path = static_cast<char **>(DMALLOC(sizeof(char*) * size, TAG_COMPILER, "compiler:init_include_path"));
+
+  // check elements and build working copy
```

Comment:
Not without local testing!!!
But you're right, that should work…
Although it probably will just replace the ```goto …``` with a ```throw(…)``` and putting the code after the label within the catch block…
(Which then very likely will be compiled into (mostly) the same instructions ;) )